### PR TITLE
fix: disallow non-existing rank indexes to be returned by `getRank`

### PR DESCRIPTION
**Pull request by DragosIorgulescu**
- may be part of the issues causing some users to experience authentication issues - as per slack thread: https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1700645470832879

## Events

N / A

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/lib/rank.ts [hunk 1]
```diff
@@ -104,7 +104,15 @@ interface GetNextRankTextProps {
...
-  return rank === 0 ? rank : rank - 1;
+  if (rank < 0) {
+    return 0;
+  }
+
+  if (rank >= RANKS.length) {
+    return RANKS.length - 1;
+  }
+
+  return rank;
```
Comment (rebelchris) [inline packages/shared/src/lib/rank.ts:115]: I think you need to do the initial calculation first, it needs to do the -1 from what I recall. (as it's based on previous rank technically)

```suggestion
const calculatedRank = rank === 0 ? rank : rank - 1
  if (calculatedRank < 0) {
    return 0;
  }

  if (calculatedRank >= RANKS.length) {
    return RANKS.length - 1;
  }

  return calculatedRank;
```

**Review by rebelchris — CHANGES_REQUESTED** [review]
Small suggestion :)

File: packages/shared/src/lib/rank.ts (hunk 1 above)
Comment (DragosIorgulescu) [reply packages/shared/src/lib/rank.ts]: ah, great catch, updated 🙏

**Review by rebelchris — APPROVED** [review]
Yep this works :)

**Pull request merged**